designs/credentialManagement/profileFileSupport.md

```diff
@@ -0,0 +1,47 @@
+# Profile File Support
+
+## Abstract
+
+This document describes the process of supporting the AWS Profile files in the [Credential Management](credentialManagement.md) system
+
+## Shared Credentials File(s)
+
+AWS SDKs and tools are capable of sharing credentials by storing them in a common location. Also known as shared credential files, the **config** and 
+**credentials** files are detailed at [CLI documentation][CliConfigDocs]. We try to comply as close as possible with the CLI credential resolution behavior, 
+but only the keys related to credential management are supported in the toolkit. 
+
+Supported keys:
+* `aws_access_key_id`
+* `aws_secret_access_key`
+* `aws_session_token`
+* `source_profile`
+* `external_id`
+* `role_session_name`
+* `mfa_serial`
+* `credential_process`
+* `sso_*` - See [SSO Document](ssoSupport.md)
+
+### File Parsing
+Parsing and merging of the **config** and **credentials** files is out of scope of this document and is handled by the Java SDK. Please see [ProfileFile](https://github.com/aws/aws-sdk-java-v2/blob/master/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFile.java).
+
+### File Locating Resolution
+Locating of the profile files is handled by the Java SDK. Please see [ProfileFileLocation](https://github.com/aws/aws-sdk-java-v2/blob/master/core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileLocation.java).
+
+### Profile File Modification Detection
+
+An Application Service (`ProfileWatcher`) is registered by the toolkit. The service instructs the IDE to notify it when the parent directory of the `config` 
+and/or `credentials` files are modified. We monitor the parent in folder in case the files do not exist yet.
+
+When the `ProfileCredentialProviderFactory` is created, it registers itself as a listener on the `ProfileWatcher`.
+
+Profiles are considered modified if, and only if, its contents (properties keys/values) modified or dependent profiles are modified. The changing of a profile name
+is considered an addition of a new profile, and deletion of the old.
+
+### Multi-Factor Authentication
+
+We support assuming a role with MFA. If the requested credential profile has the `mfa_serial` property , we will need to prompt the user for their OTP. 
+This works by blocking the `resolveCredentials` call in `AwsCredentialProvider` until a input dialog message prompt is filled in.
+
+**This has the downside of deadlocking if the UI (EDT) thread is blocking waiting on a call to AWS.**
```

Comment:
We (try to) throw an exception if this is done though?

Reply:
Sorry it is unclear. It's not talking about making a network call on the UI thread